fairlearn/reductions/grid_search/grid_search.py

```diff
@@ -52,31 +63,43 @@ def fit(self, X, Y, **kwargs):
         if self._KW_NUMBER_LAGRANGE_MULTIPLIERS in kwargs:
             number_of_lagrange_multipliers = kwargs[self._KW_NUMBER_LAGRANGE_MULTIPLIERS]
 
+        # Extract the target attribute
+        A = self._make_vector(aux_data, "aux_data")
+
+        # Extract the Y values
+        Y_vector = self._make_vector(Y, "Y")
+
+        X_rows, _ = self._get_matrix_shape(X, "X")
+        if X_rows != Y_vector.shape[0]:
+            raise RuntimeError(self._MESSAGE_X_Y_ROWS)
```

Comment:
Would appreciate thoughts on whether these (and other) exceptions are the correct ones

Reply:
Both from my experience and doing a bit of stackoverflow browsing, it seems that in cases like this both `RuntimeError` and `ValueError` are common. In the interpretability codebase, I don't think we use `RuntimeError` at all, but we are also probably too liberal with throwing around the basic `Exception`.